packages/datadog-instrumentations/src/elasticsearch.js

```diff
@@ -0,0 +1,80 @@
+'use strict'
+
+const {
+  channel,
+  addHook,
+  AsyncResource
+} = require('./helpers/instrument')
+const shimmer = require('../../datadog-shimmer')
+
+const startCh = channel('apm:elasticsearch:query:start')
+const asyncEndCh = channel('apm:elasticsearch:query:async-end')
+const endCh = channel('apm:elasticsearch:query:end')
+const errorCh = channel('apm:elasticsearch:query:error')
+
+addHook({ name: '@elastic/elasticsearch', file: 'lib/Transport.js', versions: ['>=5.6.16'] }, Transport => {
+  shimmer.wrap(Transport.prototype, 'request', wrapRequest)
+  return Transport
+})
+
+addHook({ name: 'elasticsearch', file: 'src/lib/transport.js', versions: ['>=10'] }, Transport => {
+  shimmer.wrap(Transport.prototype, 'request', wrapRequest)
+  return Transport
+})
+
+function wrapRequest (request) {
+  return function (params, options, cb) {
+    if (!startCh.hasSubscribers) {
+      return request.apply(this, arguments)
+    }
+
+    if (!params) return request.apply(this, arguments)
+
+    const asyncResource = new AsyncResource('bound-anonymous-fn')
+
+    startCh.publish([params])
```

Comment:
Why an array?

Reply:
From what I've seen so far, basically it's been to follow the convention you used for previous instrumentations to pass the arguments directly. Personally, I prefer passing an object with the minimum data we actually need in a format that is unlikely to change between versions. However, since right now everything lives in dd-trace, it doesn't really matter all that much for now.

Reply:
Yeah we should also minimize creating objects (like arrays) when there's no need to. There really isn't a need here.